Tests/MnemonicSwiftTests.swift

```diff
@@ -1,156 +1,284 @@
+        }
+
+        for testCase in testCases {
+            let expectedMnemonicString = testCase[mnenomicStringIndex]
+            let hexRepresentation = testCase[hexRepresentationIndex]
+            let mnemonicString = try Mnemonic.mnemonicString(from: hexRepresentation)
+
+            XCTAssertEqual(mnemonicString, expectedMnemonicString)
+        }
+    }
+
+    /// Test that MnemonicSwift can generate deterministic seed strings strings without a passphrase.
+    func testGenerateDeterministicSeedStringWithPassphrase() throws {
+        guard let vectors = MnemonicSwiftTests.dictionaryFromTestInputFile(),
+            let testCases = vectors[englishTestCases] as? [[String]] else {
+                XCTFail("Failed to parse input file.")
+                return
+        }
+
+        for testCase in testCases {
+            let mnemonicString = testCase[mnenomicStringIndex]
+            let expectedDeterministicSeedString = testCase[deterministicSeedStringIndex]
+
+            XCTAssertNoThrow({
+                let deterministicSeedString: String = try Mnemonic.deterministicSeedString(from: mnemonicString, passphrase: self.passphrase)
+                XCTAssertEqual(deterministicSeedString, expectedDeterministicSeedString)
+            })
+        }
+    }
+
+    static func dictionaryFromTestInputFile() -> [String: Any]? {
+        let testBundle = Bundle(for: self)
+        guard let url = testBundle.url(forResource: "vectors", withExtension: "json") else {
+            return nil
+        }
+
+        do {
+            let data = try Data(contentsOf: url)
+            let options: JSONSerialization.ReadingOptions =  [.allowFragments, .mutableContainers, .mutableLeaves]
+            guard let parsedDictionary =
+                try JSONSerialization.jsonObject(with: data, options: options) as? [String: Any] else {
+                    return nil
+            }
+            return parsedDictionary
+        } catch {
+            return nil
+        }
+    }
+
+    /// Test mnemonic generation in english.
+    func testGenerateMnemonic() {
+
+        XCTAssertNoThrow(try Mnemonic.generateMnemonic(strength: 32))
+    }
+
+    /// Prove that functions work in chinese as well.
+    func testGenerateMnemonicChinese() {
+        XCTAssertNoThrow(try Mnemonic.generateMnemonic(strength: 32, language: .chinese))
+    }
+
+    /// Test input strengths for mnemonic generation.
+    func testMnemonicGenerationStrength() {
+
+        XCTAssertNoThrow(try Mnemonic.generateMnemonic(strength: 32))
+
+        XCTAssertNoThrow(try Mnemonic.generateMnemonic(strength: 64))
+
+        XCTAssertNoThrow(try Mnemonic.generateMnemonic(strength: 128))
+
+        XCTAssertNoThrow(try Mnemonic.generateMnemonic(strength: 160))
+
+        XCTAssertNoThrow(try Mnemonic.generateMnemonic(strength: 192))
+
+        XCTAssertNoThrow(try Mnemonic.generateMnemonic(strength: 224))
+
+        XCTAssertNoThrow(try Mnemonic.generateMnemonic(strength: 256))
```

Comment:
Here we should compare that the outputs are actually the correct number of words, i.e. split on " " and make sure the resulting count correctly matches what's expected for the strength.